### PR TITLE
Add Wav2vec baseline

### PR DESCRIPTION
**Pull request by jafermarq**
Adds `wav2vec 2.0` baseline originally implemented by @tuanct1997 but now following the new format in the baselines. 

This makes #1780 #1771 obsolete

### CODE REVIEW

File: baselines/fedwav2vec2/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,142 @@
+[build-system]
+requires = ["poetry-core>=1.4.0"]
+build-backend = "poetry.masonry.api"
+
+[tool.poetry]
+name = "fedwav2vec2" # <----- Ensure it matches the name of your baseline directory containing all the source code
+version = "1.0.0"
+description = "Flower Baselines"
+license = "Apache-2.0"
+authors = ["The Flower Authors <hello@flower.dev>"]
```
Comment (jafermarq) [inline baselines/fedwav2vec2/pyproject.toml:10]: @tuanct1997 , could you extend this list with your name and email following the format above?

**Review by jafermarq — COMMENTED** [review]
Hi @tuanct1997,

Just some minor comments

File: baselines/fedwav2vec2/fedwav2vec2/conf/sb_config/w2v2.yaml [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+# Data files
+data_folder:  data/audio/TEDLIUM_release-3/data
+
+accented_letters: False
+language: en # use 'it' for Italian, 'rw' for Kinyarwanda, 'en' for english
+train_csv: data/ted_train.csv
+valid_csv: data/ted_dev.csv
+test_csv: data/ted_test.csv
+skip_prep: False # Skip data preparation
+device: cpu
```
[the remark this answers is not in the document]
Comment (tuanct1997) [reply baselines/fedwav2vec2/fedwav2vec2/conf/sb_config/w2v2.yaml:30]: @jafermarq , I think we can remove this one. In any case, we all re-write the device with cfg value from the `base.yaml`

[the remark this answers is not in the document]
Comment (jafermarq) [reply baselines/fedwav2vec2/fedwav2vec2/conf/sb_config/w2v2.yaml:30]: true. so let's remove it.

**Review by danieljanes — APPROVED** [review]

**Comment by TParcollet** [comment]
@tuanct1997 jumping randomly and way after the battle in this PR, is it expected that the sorting is set to descending? This is fairly uncomon for ASR training in SB as this feature is mostly used for debugging purpose?

**Comment by tuanct1997** [comment]
> @tuanct1997 jumping randomly and way after the battle in this PR, is it expected that the sorting is set to descending? This is fairly uncomon for ASR training in SB as this feature is mostly used for debugging purpose?

@TParcollet nice catch! You're absolutely right. Setting the sorting to descending isn't typical for ASR training in SB. I probably used that configuration during debugging and forgot to switch it back. I believe it should default to ascending in SB, which could potentially improve the performances.

Btw, yeah, that PR was quite the battlefield, wasn't it? 😄

**Pull request merged**